core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -669,6 +673,7 @@ private[spark] object JsonProtocol {
     metrics.incLocalBlocksFetched((json \ "Local Blocks Fetched").extract[Int])
     metrics.incFetchWaitTime((json \ "Fetch Wait Time").extract[Long])
     metrics.incRemoteBytesRead((json \ "Remote Bytes Read").extract[Long])
+    metrics.incRecordsRead((json \ "Records Read").extractOpt[Long].getOrElse(-1))
```

Comment:
Should this just be "0" instead of "-1". We have -1in a few places but it's for metrics which are not aggregated like this. The weird thing here is that we'll actually keep aggregating more and more negatives.